modules/gltf/src/gltf/gltf-builder.js

```diff
@@ -200,4 +195,26 @@ export default class GLTFBuilder extends GLBBuilder {
     this.json.meshes.push(glTFMesh);
     return this.json.meshes.length - 1;
   }
+
+  // Adds a binary image. Builds glTF "JSON metadata" and saves buffer reference
+  // Buffer will be copied into BIN chunk during "pack"
+  // Currently encodes as glTF image
+  addImage(imageData) {
+    const bufferViewIndex = this.addBufferView(imageData);
+    let imageIndex = null;
+
+    // Get the properties of the image to add as metadata.
+    const sizeAndType = getImageSize(imageData) || {};
+    if (sizeAndType) {
+      // width and height are non-spec fields
+      const {mimeType, width, height} = sizeAndType;
+      imageIndex = this.addImageEntry(bufferViewIndex, {mimeType, width, height});
+    } else {
+      // TODO: Spec violation.
+      //       If we are using a bufferView, mimeType must be defined
+      imageIndex = this.addImageEntry(bufferViewIndex, {});
```

Comment:
Copy the code from `addImageEntry` here and break the link between the two classes here.